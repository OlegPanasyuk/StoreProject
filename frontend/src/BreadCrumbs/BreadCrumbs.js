import React, { Component } from 'react';
import { Breadcrumb } from 'react-bootstrap';
import PropTypes from 'prop-types';

class BreadCrumbs extends Component {
    render() {
        const { obj, handleClick } = this.props;
        let arrItems = obj.map((element, i) => {
            let elem = null;
            if (i === (obj.length - 1)) {
                elem = (
                    <Breadcrumb.Item href="#" onClick={() => handleClick(element.id)} key={element.id} active>
                        {element.name}
                    </Breadcrumb.Item>
                );
            } else {
                elem = (
                    <Breadcrumb.Item href="#" onClick={() => handleClick(element.id)} key={element.id} >
                        {element.name}
                    </Breadcrumb.Item>
                );
            }
            return elem;
        });

        return (
            <Breadcrumb variant="secondary" > 
                {arrItems}
            </Breadcrumb>
        );
    }
}

BreadCrumbs.propTypes = {
    obj : PropTypes.array,
    handleClick : PropTypes.func 
};

export default BreadCrumbs;