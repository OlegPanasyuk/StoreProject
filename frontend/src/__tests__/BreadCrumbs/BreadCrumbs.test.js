import React from 'react';
import BreadCrumbs from '../../BreadCrumbs/BreadCrumbs';
import {shallow, mount, configure } from 'enzyme';
import {shallowToJson,  mountToJson} from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import { Breadcrumb } from 'react-bootstrap';
 
configure({ adapter: new Adapter() });

function setupComponent(props = {
    obj : [{
        id : null,
        name : 'Home'
    },{
        id : 1,
        name : 'Car'
    }],
    handleClick : (i) => { alert(`Click by BreadCrumb.Item${i}`);}
}) {
    let component = shallow(<BreadCrumbs {...props} />);
    return { component, props };
}


describe('BreadCrumbs Component', () => {
    
    it('Component must be created and props must be correct', () => {
        const { component, props } = setupComponent();
        const componentProps = component.props();
        
        expect(BreadCrumbs.propTypes.obj).toBe(PropTypes.array);
        expect(BreadCrumbs.propTypes.handleClick).toBe(PropTypes.func);
        expect(shallowToJson(component)).toMatchSnapshot();
    });
    
    it('Component must children must be clicked ', () => {
        const { component, props } = setupComponent();
        const componentProps = component.props();
        window.alert = jest.fn();
        for (let i = 0; i < props.obj.length; i++) {
            component.find(Breadcrumb.Item).at(i).simulate('click');
            expect(window.alert).toHaveBeenCalledWith(`Click by BreadCrumb.Item${props.obj[i].id}`);
        }
    });
});
