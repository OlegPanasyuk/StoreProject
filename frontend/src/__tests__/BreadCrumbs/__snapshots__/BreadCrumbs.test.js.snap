// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumbs Component Component must be created and props must be correct 1`] = `
<Breadcrumb
  as="nav"
  label="breadcrumb"
  listProps={Object {}}
  variant="secondary"
>
  <BreadcrumbItem
    active={false}
    as="li"
    href="#"
    key="null"
    onClick={[Function]}
  >
    Home
  </BreadcrumbItem>
  <BreadcrumbItem
    active={true}
    as="li"
    href="#"
    key="1"
    onClick={[Function]}
  >
    Car
  </BreadcrumbItem>
</Breadcrumb>
`;
